### PR TITLE
Add a global Redraw method for TextWidget

### PR DESCRIPTION
**Pull request by Seanstoppable**
Partially addresses #429, by centralizing widget drawing

### CODE REVIEW

File: modules/circleci/widget.go [hunk 1]
```diff
@@ -11,7 +11,6 @@ type Widget struct {
 	wtf.TextWidget
 	*Client
 
-	app      *tview.Application
```
Comment (Seanstoppable) [inline modules/circleci/widget.go]: Trying to eliminate this being stored here as much as possible

File: modules/gcal/widget.go [hunk 2]
```diff
@@ -61,27 +53,5 @@ func (widget *Widget) fetchAndDisplayEvents() {
 		widget.calEvents = calEvents
 	}
 
-	widget.app.QueueUpdateDraw(func() {
-		widget.display()
-	})
-}
-
-func updateLoop(widget *Widget) {
```
Comment (Seanstoppable) [inline modules/gcal/widget.go]: This seemed unnecessary, as the refresh of the widget itself should handle this, no?

File: modules/gitter/widget.go [hunk 3]
```diff
@@ -102,11 +94,10 @@ func (widget *Widget) display() {
...
 
-	widget.View.SetWrap(true)
-	widget.View.Clear()
-	widget.View.SetTitle(widget.ContextualTitle(fmt.Sprintf("%s - %s", widget.CommonSettings.Title, widget.settings.roomURI)))
-	widget.View.SetText(widget.contentFrom(widget.messages))
-	widget.View.Highlight(strconv.Itoa(widget.selected)).ScrollToHighlight()
+	title := fmt.Sprintf("%s - %s", widget.CommonSettings.Title, widget.settings.roomURI)
+
+	widget.Redraw(title, widget.contentFrom(widget.messages), true)
+	//widget.View.Highlight(strconv.Itoa(widget.selected)).ScrollToHighlight()
```
Comment (Seanstoppable) [inline modules/gitter/widget.go]: I don't have any widgets with highlight, so not 100% sure what is going on here and/or if this is a potential race condition that we should also try to abstract away

File: modules/gcal/widget.go (hunk 2 above)
Comment (senorprogrammer) [reply modules/gcal/widget.go]: Yeah, I'm not sure why that's there. It should be removed.

**Comment by senorprogrammer** [comment]
I've found a few small issues with this branch that need to be fixed first, otherwise its looking good:

* some widgets with nav keys assigned to them display the key number twice in the title
* the "new item" modal for Todo takes a long time to show up because the explicit `Draw()` that forced an update has been removed

I should be able to get to those today.

**Comment by Seanstoppable** [comment]
No problem. After this, I am going to try refactoring scrollable widgets to remove boilerplate from them too

**Comment by senorprogrammer** [comment]
All the changes from this PR have been merged https://github.com/wtfutil/wtf/commit/018d2af3ae4c17a93e6a0da4881263d6d993f577

**Pull request closed without merging**